scripts/weekly-summary

```diff
@@ -0,0 +1,374 @@
+    pass
+else:
+    signal.signal(SIGPIPE, SIG_IGN)
+
+
+def get_artifacts(token, start, end):
+    """
+    get_artifacts retrieves a list of artifact from the selected date range
+
+    It returns a list of artifact dicts like this:
+
+    {
+      "id": 187278866,
+      "node_id": "MDg6QXJ0aWZhY3QxODcyNzg4NjY=",
+      "name": "logs-daily-iso",
+      "size_in_bytes": 1047408950,
+      "url": "https://api.github.com/repos/rhinstaller/kickstart-tests/actions/artifacts/187278866",
+      "archive_download_url": "https://api.github.com/repos/rhinstaller/kickstart-tests/actions/artifacts/187278866/zip",
+      "expired": false,
+      "created_at": "2022-03-17T03:30:12Z",
+      "updated_at": "2022-03-17T03:30:13Z",
+      "expires_at": "2022-03-25T03:29:06Z"
+    },
+    """
+    req = Request(URL)
+    req.add_header("Accept", "application/vnd.github.v3+json")
+    req.add_header("Authorization", f"token {token}")
+    with urlopen(req) as r:
+        data = json.load(r)
+
+    # Filter out the artifacts within the date range and names
+    artifacts = []
+    for a in data["artifacts"]:
+        if a["name"] not in ARTIFACTS:
+            continue
+        updated_at = datetime.fromisoformat(a["updated_at"][:-1])
+        if start <= updated_at <= end:
+            artifacts.append(a)
+
+    return sorted(artifacts, key=lambda x: x["updated_at"])
+
+
+def run_curl(token, url, filename):
+    """
+    run_curl downloads an artifact file
+    """
+    with open(filename, "wb") as f:
+        c = pycurl.Curl()
+        headers = [
+           "Accept: application/vnd.github.v3+json",
+           f"Authorization: token {token}",
+        ]
+
+        options = {
+            pycurl.FOLLOWLOCATION:  1,
+            pycurl.MAXREDIRS:       5,
+            pycurl.CONNECTTIMEOUT:  30,
+            pycurl.TIMEOUT:         300,
+            pycurl.NOSIGNAL:        1,
+            pycurl.URL:             url,
+            pycurl.HTTPHEADER:      headers,
+            pycurl.WRITEDATA:       f
+        }
+        for k, v in options.items():
+            c.setopt(k, v)
+
+        try:
+            c.perform()
+            ok = (True, None)
+        except Exception as e:
+            print(f"ERROR: {e}", file=sys.stderr)
+            sys.stderr.flush()
+            ok = (False, e)
+        c.close()
+
+    return ok
+
+
+def download_artifacts(token, artifacts):
+    """
+    download_artifacts downloads the artifacts as uniquely named files
+    """
+    zipfiles = []
+    for a in artifacts:
+        updated_at = datetime.fromisoformat(a["updated_at"][:-1])
+        datename = a["name"]+updated_at.strftime("-%Y-%m-%d")
+        filename = datename + ".zip"
+        zipfiles.append((a["name"], datename, filename))
+        if os.path.exists(filename):
+            print(f"{filename} skipped, already downloaded")
+            continue
+
+        print(f"Fetching {filename}")
+        run_curl(token, a["archive_download_url"], filename)
```

Comment:
Maybe the return value could be checked here but reacting to it would complicate the things quite a lot at this stage so it can be eventually done when / if we really hit some failures.

Reply:
right, run_curl actually logs the errors, and I want to continue with the other logs so there is no need to check, even though it does return the status.